### ISSUE TITLE
Magic resistance inconsistencies

### ISSUE THREAD

**Issue by Block-Vader** [00-Evan/shattered-pixel-dungeon#713]
This might not be a bug but I still think it should be adressed. Scrolls and enchantments are considered magical (have no effect if the Hero has magic immunity), but glyph of antimagic and other forms of magic damage reduction offer no protection against them. The same is true for cursed wands, which can directly hurt target/user with explosion or health transfer.

**Comment by 00-Evan** [comment]
I'm aware that magic resist has several oddities, though some of the ones you mentioned are correct, e.g. explosions are physical damage, even if they were created by magic. I'll keep this issue open but I'm deferring it for now as going over all of the various effects isn't a priority atm.

**Comment by 00-Evan** [comment]
Fixed in commit 2644e16d4cce5974d032bc31fc2abc20fd64b970

**Comment by Block-Vader** [comment]
This doesn't seem right. The commit is "fixed magic immune not applying to rings or artifacts", it has nothing to do with magic resistance.

**Comment by 00-Evan** [comment]
Apologies, there were two commits and I only referenced one. Second one is 59d8e810525c3fba525788ba4854538d0d1788bc

As I mentioned before, some of the effects you mentioned above are working correctly, e.g. most explosions are physical damage even if they are conjured from a magical source. Psi blast and magical bombs were missing from resists though, and so I added them. It does seem like I've missed a couple cursed wand effects though (at lest in some cases, often it is correctly resisted), so I'll look into that for v1.4.1
